python/cudf/cudf/core/dataframe.py

```diff
@@ -2619,39 +2623,65 @@ def columns(self):
     @columns.setter  # type: ignore
     @_cudf_nvtx_annotate
     def columns(self, columns):
-        if isinstance(columns, cudf.BaseIndex):
-            columns = columns.to_pandas()
-        if columns is None:
-            columns = pd.Index(range(len(self._data.columns)))
-        is_multiindex = isinstance(columns, pd.MultiIndex)
-
-        if isinstance(columns, (Series, cudf.Index, ColumnBase)):
-            columns = pd.Index(columns.to_numpy(), tupleize_cols=is_multiindex)
-        elif not isinstance(columns, pd.Index):
-            columns = pd.Index(columns, tupleize_cols=is_multiindex)
+        multiindex = False
+        rangeindex = False
+        label_dtype = None
```

Comment:
This will end up None for the MultiIndex case. Is that expected?

Reply:
Yup. `label_dtype` is used for `pandas.Index(..., dtype=label_dtype)` in `to_pandas_index`. We currently have no similar mechanism for the MultiIndex case